lib/region/europe/index.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * @module Module for determining if a locale is in the EU.
+ *
+ * Examples:
+ *
+ * // Example 1, loading this module directly:
+ * var europe = require('bv-ui-core/lib/region/europe');
+ * europe.hasLocale('en_US'); // false
+ * europe.hasLocale('en_GB'); // true
+ *
+ * europe.hasTerritory('US'); // false
+ * europe.hasTerritory('GB'); // true
+ *
+ * // Example 2, loading the region module:
+ * var region = require('bv-ui-core/lib/region');
+ * region.hasLocale('europe', 'en_US'); // false
+ * region.hasLocale('europe', 'en_GB'); // true
+ *
+ * region.hasTerritory('europe', 'US'); // false
+ * region.hasTerritory('europe', 'GB'); // true
+ */
+
+// territories is a map that contains an entry for each valid EU country
+// code, so territories[territory] will return either true or undefined.
+// We just need to coerce that into a boolean value and we're good to go.
+var territories = require('./territories');
+
+// Some constant patterns for valid values.
+var VALID_TERRITORY = /^[A-Z][A-Z]$/;
+var VALID_LOCALE = /^[a-z][a-z][\_\-][A-Z][A-Z]$/;
```

Comment:
Some basic simplifications:

```javascript
var VALID_TERRITORY = /^[A-Z]{2}$/;
var VALID_LOCALE = /^[a-z]{2}[-_][A-Z]{2}$/;
```

As regex goes, I feel like it makes them a bit clearer. The `[-_]` part works as `[_-]` as well, and neither character requires escaping. The `-` only has special meaning to imply ranges when it makes sense, like it does in `[a-z]`.